PowerShellTools/Intellisense/BraceCompletionController.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+using EnvDTE;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Repl;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace PowerShellTools.Intellisense
+{
+    /// <summary>
+    /// Command handler in the command chain that is used for complete braces for both the editor and REPL window.
+    /// </summary>
+    internal sealed class BraceCompletionController : IOleCommandTarget

```

Comment:
Since you covered quite a few brace completion scenario here, please add corresponding unit test for them, otherwise, it'll be easy to break when others start modifying the code here.